plugin/src/main/java/org/opensearch/ml/plugin/MachineLearningPlugin.java

```diff
@@ -345,6 +368,13 @@ public Collection<Object> createComponents(
             nodeHelper
         );
 
+        try {
+            CompletableFuture<Void> completableFuture = CompletableFuture.runAsync(() -> mlModelAndNodeManager.autoReLoadModel());
+            completableFuture.get();
```

Comment:
Do not use blocking approach in our code, always use callbacks to not blocking the main thread.

Reply:
move in ModelAutoReloader class